src/App.tsx

```diff
@@ -36,12 +36,28 @@ const App = (): JSX.Element => {
     const addPurchase = (purchase: IPurchase) => {
         setPurchaseList([...purchaseList, purchase])
     }
+    const editPurchase = (purchase: IPurchase) => {
+        purchase.isEdit = !purchase.isEdit
+        setPurchaseList([...purchaseList])
+    }
```

Comment:
You are trying to access the object which is linked to the object in the state and change it - which is bad. You need to create a new array, change the property of one of its' element and then call setPurchaseList(<newArray>)